impurityModel/test/test_comparison_with_reference.py

```diff
@@ -11,50 +13,58 @@
 import h5py
 import numpy as np
 
+DIR_PATH = os.path.dirname(os.path.realpath(__file__))
+
+SCRIPT_PATH = os.path.join(DIR_PATH, "../../scripts/run_Ni_NiO_Xbath.sh")
+REFERENCE_SPECTRA_PATH = os.path.join(DIR_PATH, "referenceOutput/Ni_NiO_50bath/spectra.h5")
+
 
 def test_comparison():
     compare_spectra()
 
 
 def compare_spectra(
-    script_file="scripts/run_Ni_NiO_Xbath.sh",
+    script_path=SCRIPT_PATH,
     script_argument=50,
-    reference_file="referenceOutput/Ni_NiO_50bath/spectra.h5",
+    reference_spectra_path=REFERENCE_SPECTRA_PATH,
 ):
-    print("Start comparison of spectra...")
     # Create a temporary directory using the context manager
     with tempfile.TemporaryDirectory() as tmpdirname:
         print("Created temporary directory", tmpdirname)
         os.chdir(tmpdirname)
         print("Current working dir:", os.getcwd())
-        path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
-        print(f"{path = }")
-        cmd = os.path.join(path[:-19], script_file)
-        print("Run command:", cmd)
-        print("Use command argument:", script_argument)
+        print(f"{script_path = }")
+        print(f"{script_argument = }")
 
-        subprocess.run(args=[cmd, str(script_argument)], stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        output = subprocess.run(args=[script_path, str(script_argument)], check=False, capture_output=True)
+        print(f"{output.returncode = }")
+        print("output.stdout:")
+        print(str(output.stdout, encoding="utf-8"))
+        print("output.stderr:")
+        print(str(output.stderr, encoding="utf-8"))
+        assert output.returncode == 0
```

Comment:
```suggestion
        output = subprocess.run(args=[script_path, str(script_argument)], check=True)
```

You can get stdout and stderr printed by default and assert on the returncode for free, from the subprocess module.